### ISSUE TITLE
Build error with GCC 10

### ISSUE THREAD

**Issue by bertwesarg** [dash-project/dash#704]
I get the following error with GCC 10 on f84e89ecf:

```
In file included from /home/wesarg/opt/gcc-10.1/include/c++/10.1.0/bits/locale_conv.h:41,
                 from /home/wesarg/opt/gcc-10.1/include/c++/10.1.0/locale:43,
                 from /home/wesarg/opt/gcc-10.1/include/c++/10.1.0/iomanip:43,
                 from /home/wesarg/Work/Mephisto/dash/dash/include/dash/internal/Logging.h:17,
                 from /home/wesarg/Work/Mephisto/dash/dash/include/dash/Exception.h:12,
                 from /home/wesarg/Work/Mephisto/dash/dash/include/dash/Team.h:8,
                 from /home/wesarg/Work/Mephisto/dash/dash/include/libdash.h:27,
                 from /home/wesarg/Work/Mephisto/dash/dash/examples/bench.14.accumulate/main.cpp:6:
/home/wesarg/opt/gcc-10.1/include/c++/10.1.0/bits/unique_ptr.h: In instantiation of 'std::__uniq_ptr_impl<_Tp, _Dp>::__uniq_ptr_impl(std::__uniq_ptr_impl<_Tp, _Dp>&&) [with _Tp = dash::Atomic<int>; _Dp = dash::DefaultDeleter<dash::GlobalAllocator<dash::Atomic<int>, dash::GlobStaticMem<dash::HostSpace> > >]':
/home/wesarg/opt/gcc-10.1/include/c++/10.1.0/bits/unique_ptr.h:210:7:   required from 'auto dash::allocate_unique(const AllocatorT&, std::size_t) [with T = dash::Atomic<int>; AllocatorT = dash::GlobalAllocator<dash::Atomic<int>, dash::GlobStaticMem<dash::HostSpace> >; Deleter = dash::DefaultDeleter; std::size_t = long unsigned int]'
/home/wesarg/Work/Mephisto/dash/dash/include/dash/Array.h:723:40:   required from 'class dash::Array<dash::Atomic<int> >'
/home/wesarg/Work/Mephisto/dash/dash/include/dash/Coevent.h:49:39:   required from here
/home/wesarg/opt/gcc-10.1/include/c++/10.1.0/bits/unique_ptr.h:162:22: error: no match for 'operator=' (operand types are 'std::__uniq_ptr_impl<dash::Atomic<int>, dash::DefaultDeleter<dash::GlobalAllocator<dash::Atomic<int>, dash::GlobStaticMem<dash::HostSpace> > > >::pointer' {aka 'std::allocator_traits<dash::GlobalAllocator<dash::Atomic<int>, dash::GlobStaticMem<dash::HostSpace> > >::pointer'} and 'std::nullptr_t')
  162 |       { __u._M_ptr() = nullptr; }
      |         ~~~~~~~~~~~~~^~~~~~~~~
In file included from /home/wesarg/Work/Mephisto/dash/dash/include/dash/iterator/GlobIter.h:4,
                 from /home/wesarg/Work/Mephisto/dash/dash/include/dash/iterator/IteratorTraits.h:9,
                 from /home/wesarg/Work/Mephisto/dash/dash/include/dash/Iterator.h:6,
                 from /home/wesarg/Work/Mephisto/dash/dash/include/libdash.h:31,
                 from /home/wesarg/Work/Mephisto/dash/dash/examples/bench.14.accumulate/main.cpp:6:
/home/wesarg/Work/Mephisto/dash/dash/include/dash/GlobPtr.h:122:11: note: candidate: 'constexpr dash::GlobPtr<ElementType, MemorySpace>::self_t& dash::GlobPtr<ElementType, MemorySpace>::operator=(const self_t&) [with ElementType = dash::Atomic<int>; GlobMemT = dash::GlobStaticMem<dash::HostSpace>; dash::GlobPtr<ElementType, MemorySpace>::self_t = dash::GlobPtr<dash::Atomic<int>, dash::GlobStaticMem<dash::HostSpace> >]'
  122 |   self_t &operator=(const self_t &rhs) = default;
      |           ^~~~~~~~
/home/wesarg/Work/Mephisto/dash/dash/include/dash/GlobPtr.h:122:35: note:   no known conversion for argument 1 from 'std::nullptr_t' to 'const self_t&' {aka 'const dash::GlobPtr<dash::Atomic<int>, dash::GlobStaticMem<dash::HostSpace> >&'}
  122 |   self_t &operator=(const self_t &rhs) = default;
      |                     ~~~~~~~~~~~~~~^~~
/home/wesarg/Work/Mephisto/dash/dash/include/dash/GlobPtr.h:159:11: note: candidate: 'constexpr dash::GlobPtr<ElementType, MemorySpace>::self_t& dash::GlobPtr<ElementType, MemorySpace>::operator=(dash::GlobPtr<ElementType, MemorySpace>::self_t&&) [with ElementType = dash::Atomic<int>; GlobMemT = dash::GlobStaticMem<dash::HostSpace>; dash::GlobPtr<ElementType, MemorySpace>::self_t = dash::GlobPtr<dash::Atomic<int>, dash::GlobStaticMem<dash::HostSpace> >]'
  159 |   self_t &operator=(self_t &&rhs) DASH_NOEXCEPT = default;
      |           ^~~~~~~~
/home/wesarg/Work/Mephisto/dash/dash/include/dash/GlobPtr.h:159:30: note:   no known conversion for argument 1 from 'std::nullptr_t' to 'dash::GlobPtr<dash::Atomic<int>, dash::GlobStaticMem<dash::HostSpace> >::self_t&&' {aka 'dash::GlobPtr<dash::Atomic<int>, dash::GlobStaticMem<dash::HostSpace> >&&'}
  159 |   self_t &operator=(self_t &&rhs) DASH_NOEXCEPT = default;
      |                     ~~~~~~~~~^~~
```

Though this may also be a compiler error.

**Comment by dhinf** [comment]
same with clang 10

**Comment by devreal** [comment]
Looks like we use `std::unique_ptr<dash::GlobPtr<dash::Atomic<T>>>` somewhere and that fails because we canbot assign `nullptr` to `dash::GlobPtr`?

**Comment by dhinf** [comment]
correct. i'm looking into it, because nothing compiles at the moment with gcc and clang 10

**Comment by dhinf** [comment]
i found a solution and will open a pull request soon

**Comment by dhinf** [comment]
fixed with pr #710

**Comment by devreal** [comment]
Fixed with #710. Closing.
